src/rate_limiter.cc

```diff
@@ -107,8 +107,11 @@ RateLimiter::UnregisterModel(const TritonModel* model)
     RETURN_IF_ERROR(resource_manager_->UpdateResourceLimits());
   }
 
-  if (payload_queues_.find(model) != payload_queues_.end()) {
-    payload_queues_.erase(model);
+  {
+    std::lock_guard<std::mutex> lk1(payload_queues_mu_);
```

Comment:
just `lk`

Reply:
Should also add lock on creating the payload_queue

Reply:
Thanks, updated and added.